src/common/security/secret/context.go

```diff
@@ -114,7 +89,9 @@ func (s *SecurityContext) GetMyProjects() ([]*models.Project, error) {
 // GetProjectRoles return guest role if has read permission, otherwise return nil
 func (s *SecurityContext) GetProjectRoles(projectIDOrName interface{}) []int {
 	roles := []int{}
-	if s.HasReadPerm(projectIDOrName) {
+	if s.store != nil &&
```

Comment:
I think we need to refactor to implement a different security Context for solution users.

Reply:
ISSUE https://github.com/goharbor/harbor/issues/6882